openeogeotrellis/backend.py

```diff
@@ -1899,46 +1901,58 @@ def get_results(self, job_id: str, user_id: str) -> Dict[str, dict]:
             ml_model_metadata['asset'] = False
             results_dict['ml_model_metadata.json'] = ml_model_metadata
 
-        if os.path.isfile(job_dir / 'out'):
-            results_dict['out'] = {
-                # TODO: give meaningful filename and extension
-                "asset": True,
-                "output_dir": str(job_dir),
-                "type": media_type,
-                "bands": bands,
-                "nodata": nodata
-            }
-
-        if os.path.isfile(job_dir / 'profile_dumps.tar.gz'):
-            results_dict['profile_dumps.tar.gz'] = {
-                "asset": True,
-                "output_dir": str(job_dir),
-                "type": "application/gzip"
-            }
-
-        for file_name in os.listdir(job_dir):
-            if file_name.endswith("_metadata.json") and file_name != JOB_METADATA_FILENAME:
-                results_dict[file_name] = {
+        # Updating the asset info based on what is on the file system was a solution for some old edge cases.
```

Comment:
"old edge cases" sounds like a legacy, obsolete system. But isn't this "non-kubernetes" way the standard way on the VITO YARN deploy, which is all but obsolete

Reply:
This comment turned out to be inaccurate, it is still in use quite a bit. 

Moved away from using is_kube_deploy here.

For the if right below, it was more appropriate to check if the job directory exists. 
https://github.com/Open-EO/openeo-geopyspark-driver/blob/1c2d0be62d822ba4e383009ee86b30ea529e62e0/openeogeotrellis/backend.py#L1921-L1923

And in the for at the bottom, we check with ConfigParams().use_object_storage to see if we should set the output_dir here, or leave it alone because it should have a value already, pointing to S3.
https://github.com/Open-EO/openeo-geopyspark-driver/blob/1c2d0be62d822ba4e383009ee86b30ea529e62e0/openeogeotrellis/backend.py#L1961-L1963